GitUI/CommandsDialogs/FormAddToGitIgnore.cs

```diff
@@ -70,25 +70,23 @@ private void AddToIgnoreClick(object sender, EventArgs e)
 
             try
             {
-                var fileName = ExcludeFile;
-                Validates.NotNull(fileName);
-                FileInfoExtensions.MakeFileTemporaryWritable(fileName, x =>
+                FileInfoExtensions.MakeFileTemporaryWritable(Validates.NotNull(ExcludeFile), fileName =>
                 {
                     StringBuilder gitIgnoreFileAddition = new();
 
                     if (File.Exists(fileName) && !File.ReadAllText(fileName, GitModule.SystemEncoding).EndsWith(Environment.NewLine))
                     {
-                        gitIgnoreFileAddition.Append(Environment.NewLine);
+                        gitIgnoreFileAddition.AppendLine();
```

Comment:
This is not the same, AppenLine() will add `\r\n` and Git expects `\n` even if I believe `\r\n` should work.
But next tool that update may convert the line ending and reviewers will complain.

Reply:
`Append(Environment.NewLine)` has always been equal to `AppendLine()`. I have verified it yet another time.

I have not found an explicit statement in the git docs about what line endings shall be used for this **local** file `.git/info/exclude`.

> Since this change attracted so many comments we'll need tests to ensure the end result is expected and git compliant.

I was already afraid that I will regret this simple refactoring of this ugly code, which does correctly handle only the platform-native line-endings. So I am going to remove the unrelated changes and leave it open for a follow-up.

Reply:
It's not that you are doing either environment.newline or appendline. It's that we are causing a file that is understood by git to be \n separated even if environment.newline = \r\n.